lib/core/src/Cardano/Wallet/Transaction.hs

```diff
@@ -168,6 +175,37 @@ defaultTransactionCtx = TransactionCtx
 data DelegationAction = RegisterKeyAndJoin PoolId | Join PoolId | Quit
     deriving (Show, Eq, Generic)
 
+-- | Indicates a problem with the selection criteria for a coin selection.
+data ErrSelectionCriteria
+    = ErrSelectionCriteriaOutputTokenBundleSizeExceedsLimit
+        ErrTokenBundleSizeExceedsLimit
+    | ErrSelectionCriteriaOutputTokenQuantityExceedsMaxBound
+        ErrTokenQuantityExceedsMaxBound
+    deriving (Eq, Generic, Show)
+
+data ErrTokenBundleSizeExceedsLimit = ErrTokenBundleSizeExceedsLimit
+    { address :: !Address
+      -- ^ The address to which this token bundle was to be sent.
+    , assetCount :: !Int
+      -- ^ The number of assets within the token bundle.
```

Comment:
Note that it might be possible to be more helpful here, by giving a suggestion of **_how many bundles_** the user will need to split their bundle into.

But this is harder than it sounds, as different partitions give rise to different serialized lengths. Consider this example:

- Bundle **A**:
    - `100` assets with names of length `1`
    - `100` assets with names of length `32`.

We might determine that it's feasible to split this into 2 valid bundles, if the bundles are partitioned this way:

- Bundle **A1**:
    - `50` assets with names of length `1`
    - `50` assets with names of length `32`.
- Bundle **A2**:
    - `50` assets with names of length `1`
    - `50` assets with names of length `32`.

But if the user is not careful, they might end up partitioning the bundles like this:

- Bundle **A3**:
    - `100` assets with names of length `1`.
- Bundle **A4**:
    - `100` assets with names of length `32`.

In the above example, bundle **A4** is still likely to be rejected.

Reply:
Another option (for the future) would be to provide a suggested partitioning in the API error response, or to provide automated splitting of output bundles (should the user request this).